Lib/fractions.py

```diff
@@ -115,15 +115,16 @@ def __new__(cls, numerator=0, denominator=None, *, _normalize=True):
                     raise ValueError('Invalid literal for Fraction: %r' %
                                      numerator)
                 numerator = int(m.group('num') or '0')
-                denom = m.group('denom')
-                if denom:
-                    denominator = int(denom)
+                denominator = m.group('den')
+                if denominator:
+                    denominator = int(denominator)
                 else:
                     denominator = 1
                     decimal = m.group('decimal')
                     if decimal:
-                        scale = 10**len(decimal)
-                        numerator = numerator * scale + int(decimal)
+                        decimal = int(decimal)
```

Comment:
This loses information about leading zeros in the fractional part, leading to this bug:

```
>>> from fractions import Fraction
>>> Fraction("1.01")
Fraction(11, 10)
```

I think this set of changes can be reverted.

Reply:
I doubt that can be reverted, but I think it's now corrected.